core/controllers/acl_decorators.py

```diff
@@ -2181,11 +2181,17 @@ def test_can_access(self, **kwargs):
 
         Raises:
             NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have
+                credentials to access the page.
         """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
         if role_services.ACTION_ACCESS_LEARNER_DASHBOARD in self.user.actions:
             return handler(self, **kwargs)
         else:
-            raise self.NotLoggedInException
+            raise self.UnauthorizedUserException(
+                'You do not have the credentials to access this page.')
```

Comment:
I have made this change because I think if someone is not able to access a page then we should throw `UnauthorizedUserException` rather than `NotLoggedInException`.
I can revert it if this change isn't required.

Reply:
This change seems reasonable. Thanks!

Reply:
Great!